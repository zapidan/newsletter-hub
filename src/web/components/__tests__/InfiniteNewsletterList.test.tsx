import { TestProviders } from '@common/components/providers/AppProviders';
import { NewsletterWithRelations } from '@common/types';
import { render, screen } from '@testing-library/react';
import { vi } from 'vitest';
import { InfiniteNewsletterList } from '../InfiniteScroll/InfiniteNewsletterList';

// Mock the date formatting function to ensure consistent snapshots
vi.mock('@/common/utils/date', () => ({
  formatDate: vi.fn().mockImplementation((date) => `Formatted: ${new Date(date).toISOString()}`),
  formatRelativeTime: vi.fn().mockReturnValue('2 days ago'),
}));

describe('InfiniteNewsletterList', () => {
  // Mock system time to a fixed date for consistent snapshots
  beforeAll(() => {
    vi.useFakeTimers();
    vi.setSystemTime(new Date('2023-01-03T00:00:00Z'));
  });

  afterAll(() => {
    vi.useRealTimers();
  });

  const mockNewsletters: NewsletterWithRelations[] = [
    {
      id: '1',
      title: 'Test Newsletter 1',
      content: 'This is a test newsletter',
      summary: 'Test summary',
      image_url: 'https://example.com/image.jpg',
      received_at: '2023-01-01T00:00:00Z',
      updated_at: '2023-01-01T00:00:00Z',
      is_read: false,
      is_liked: false,
      is_archived: false,
      user_id: 'user-1',
      newsletter_source_id: 'source-1',
      source_id: 'source-1',
      source: {
        id: 'source-1',
        name: 'Test Source',
        from: 'test@example.com',
        user_id: 'user-1',
        created_at: '2023-01-01T00:00:00Z',
        updated_at: '2023-01-01T00:00:00Z',
        is_archived: false
      },
      tags: [],
      word_count: 100,
      estimated_read_time: 1
    },
    {
      id: '2',
      title: 'Test Newsletter 2',
      content: 'Another test newsletter',
      summary: 'Test summary',
      image_url: 'https://example.com/image.jpg',
      received_at: '2023-01-02T00:00:00Z',
      updated_at: '2023-01-02T00:00:00Z',
      is_read: true,
      is_liked: false,
      is_archived: false,
      user_id: 'user-1',
      newsletter_source_id: 'source-2',
      source_id: 'source-2',
      source: {
        id: 'source-2',
        name: 'Test Source 2',
        from: 'test2@example.com',
        user_id: 'user-1',
        created_at: '2023-01-02T00:00:00Z',
        updated_at: '2023-01-02T00:00:00Z',
        is_archived: false
      },
      tags: [],
      word_count: 100,
      estimated_read_time: 1
    },
  ];

  it('renders loading state correctly', () => {
    const { container } = render(
      <TestProviders>
        <InfiniteNewsletterList
          newsletters={[]}
          isLoading={true}
          isLoadingMore={false}
          hasNextPage={false}
          totalCount={0}
          error={null}
          onLoadMore={() => { }}
          onToggleLike={async () => { }}
          onRowClick={() => { }}
        />
      </TestProviders>
    );

    expect(container).toMatchSnapshot();
    expect(container.querySelector('.animate-spin')).toBeInTheDocument();
    expect(screen.getByText('Loading newsletters...')).toBeInTheDocument();
  });

  it('renders empty state when no newsletters are available', () => {
    const { container } = render(
      <TestProviders>
        <InfiniteNewsletterList
          newsletters={[]}
          isLoading={false}
          isLoadingMore={false}
          hasNextPage={false}
          totalCount={0}
          error={null}
          onLoadMore={() => { }}
          onToggleLike={async () => { }}
          onRowClick={() => { }}
        />
      </TestProviders>
    );

    expect(container).toMatchSnapshot();
    expect(screen.getByText(/no newsletters found/i)).toBeInTheDocument();
  });

  it('renders newsletters list correctly', () => {
    const { container } = render(
      <TestProviders>
        <InfiniteNewsletterList
          newsletters={mockNewsletters}
          isLoading={false}
          isLoadingMore={false}
          hasNextPage={false}
          totalCount={mockNewsletters.length}
          error={null}
          onLoadMore={() => { }}
          onToggleLike={async () => { }}
          onRowClick={() => { }}
        />
      </TestProviders>
    );

    expect(container).toMatchSnapshot();
    expect(screen.getByText('Test Newsletter 1')).toBeInTheDocument();
    expect(screen.getByText('Test Newsletter 2')).toBeInTheDocument();
  });

  it('shows loading more indicator when fetching next page', () => {
    const { container } = render(
      <TestProviders>
        <InfiniteNewsletterList
          newsletters={mockNewsletters}
          isLoading={false}
          isLoadingMore={true}
          hasNextPage={true}
          totalCount={mockNewsletters.length}
          error={null}
          onLoadMore={() => { }}
          onToggleLike={async () => { }}
          onRowClick={() => { }}
        />
      </TestProviders>
    );

    // Verify the loading spinner is present with the correct classes
    const loadingSpinner = container.querySelector('.animate-spin');
    expect(loadingSpinner).toBeInTheDocument();

    // Check for the SVG element and its classes
    const svgElement = loadingSpinner?.closest('svg');
    expect(svgElement).toHaveClass('w-6', 'h-6', 'text-blue-500');

    // Verify the loading text is present
    expect(screen.getByText('Loading more newsletters...')).toBeInTheDocument();

    // Verify the snapshot matches
    expect(container).toMatchSnapshot();
  });

  it('calls onNewsletterClick when a newsletter is clicked', () => {
    const handleClick = vi.fn();

    render(
      <TestProviders>
        <InfiniteNewsletterList
          newsletters={mockNewsletters}
          isLoading={false}
          isLoadingMore={false}
          hasNextPage={false}
          totalCount={mockNewsletters.length}
          error={null}
          onLoadMore={() => { }}
          onToggleLike={async () => { }}
          onRowClick={handleClick}
        />
      </TestProviders>
    );

    screen.getByText('Test Newsletter 1').click();
    expect(handleClick).toHaveBeenCalledWith(mockNewsletters[0]);
  });
});
