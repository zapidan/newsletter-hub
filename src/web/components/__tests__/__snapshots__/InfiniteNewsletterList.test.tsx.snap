// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`InfiniteNewsletterList > renders empty state when no newsletters are available 1`] = `
<div>
  <div
    class="flex flex-col items-center justify-center py-16 text-center "
  >
    <div
      class="text-gray-400 mb-4"
    >
      <svg
        class="w-16 h-16 mx-auto"
        fill="none"
        stroke="currentColor"
        viewBox="0 0 24 24"
      >
        <path
          d="M20 13V6a2 2 0 00-2-2H6a2 2 0 00-2 2v7m16 0v5a2 2 0 01-2 2H6a2 2 0 01-2 2m16-7H4m16 0l-2-2m-12 2l2-2"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
        />
      </svg>
    </div>
    <h3
      class="text-lg font-medium text-gray-900 mb-2"
    >
      No newsletters found
    </h3>
    <p
      class="text-sm text-gray-500"
    >
      No newsletters match your current filters. Try adjusting your search criteria.
    </p>
  </div>
</div>
`;

exports[`InfiniteNewsletterList > renders loading state correctly 1`] = `
<div>
  <div
    class="flex flex-col items-center justify-center py-16 "
  >
    <div
      class="animate-spin rounded-full h-12 w-12 border-t-2 border-b-2 border-blue-500 mb-4"
    />
    <p
      class="text-base text-neutral-400"
    >
      Loading newsletters...
    </p>
  </div>
</div>
`;

exports[`InfiniteNewsletterList > renders newsletters list correctly 1`] = `
<div>
  <div
    class=""
  >
    <div
      class="space-y-0 divide-y divide-gray-100"
    >
      <div
        class="rounded-lg p-4 flex items-start cursor-pointer transition-all duration-200 bg-blue-50/60 border-l-3 border-blue-500 hover:bg-blue-100/50  border border-neutral-200"
      >
        <div
          class="flex-1 min-w-0"
        >
          <div
            class="flex items-start gap-3 mb-1"
          >
            <img
              alt="Test Newsletter 1"
              class="w-10 h-10 rounded object-cover bg-gray-100 flex-shrink-0 mt-1"
              src="https://example.com/image.jpg"
            />
            <div
              class="flex-1 min-w-0"
            >
              <div
                class="flex items-start justify-between gap-3"
              >
                <div
                  class="flex-1 min-w-0"
                >
                  <div
                    class="font-semibold text-base truncate"
                  >
                    Test Newsletter 1
                  </div>
                  <div
                    class="text-sm text-gray-500 truncate"
                  >
                    Test Source
                    <span
                      class="text-gray-400 ml-2"
                    >
                      • 
                      test@example.com
                    </span>
                  </div>
                </div>
                <div
                  class="flex items-center gap-1 flex-shrink-0"
                >
                  <button
                    class="btn btn-ghost btn-xs p-1.5 rounded-lg hover:bg-gray-200 transition-colors "
                    title="Edit tags"
                    type="button"
                  >
                    <svg
                      class="lucide lucide-tag text-gray-500 hover:text-primary-600"
                      fill="none"
                      height="14"
                      stroke="currentColor"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="2"
                      viewBox="0 0 24 24"
                      width="14"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M12.586 2.586A2 2 0 0 0 11.172 2H4a2 2 0 0 0-2 2v7.172a2 2 0 0 0 .586 1.414l8.704 8.704a2.426 2.426 0 0 0 3.42 0l6.58-6.58a2.426 2.426 0 0 0 0-3.42z"
                      />
                      <circle
                        cx="7.5"
                        cy="7.5"
                        fill="currentColor"
                        r=".5"
                      />
                    </svg>
                  </button>
                  <div
                    class="flex items-center gap-0.5"
                  >
                    <button
                      class="btn btn-ghost btn-xs p-1.5 rounded-lg hover:bg-gray-200 transition-all "
                      title="Mark as read"
                      type="button"
                    >
                      <svg
                        class="lucide lucide-eye h-3.5 w-3.5 text-blue-600"
                        fill="none"
                        height="24"
                        stroke="currentColor"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-width="2"
                        viewBox="0 0 24 24"
                        width="24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M2 12s3-7 10-7 10 7 10 7-3 7-10 7-10-7-10-7Z"
                        />
                        <circle
                          cx="12"
                          cy="12"
                          r="3"
                        />
                      </svg>
                    </button>
                    <button
                      class="btn btn-ghost btn-xs p-1.5 rounded-lg hover:bg-gray-200 transition-all "
                      title="Like"
                      type="button"
                    >
                      <svg
                        class="lucide lucide-heart h-3.5 w-3.5 text-gray-600 hover:text-red-500"
                        fill="none"
                        height="24"
                        stroke="currentColor"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-width="2"
                        viewBox="0 0 24 24"
                        width="24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M19 14c1.49-1.46 3-3.21 3-5.5A5.5 5.5 0 0 0 16.5 3c-1.76 0-3 .5-4.5 2-1.5-1.5-2.74-2-4.5-2A5.5 5.5 0 0 0 2 8.5c0 2.3 1.5 4.05 3 5.5l7 7Z"
                        />
                      </svg>
                    </button>
                    <button
                      class="btn btn-ghost btn-xs p-1.5 rounded-lg hover:bg-gray-200 transition-all "
                      title="Add to queue"
                      type="button"
                    >
                      <svg
                        class="lucide lucide-bookmark h-3.5 w-3.5 text-gray-600 hover:text-yellow-500"
                        fill="none"
                        height="24"
                        stroke="currentColor"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-width="2"
                        viewBox="0 0 24 24"
                        width="24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="m19 21-7-4-7 4V5a2 2 0 0 1 2-2h10a2 2 0 0 1 2 2v16z"
                        />
                      </svg>
                    </button>
                    <button
                      class="btn btn-ghost btn-xs p-1.5 rounded-lg hover:bg-gray-200 transition-all "
                      title="Archive"
                      type="button"
                    >
                      <svg
                        class="lucide lucide-archive h-3.5 w-3.5 text-amber-700"
                        fill="none"
                        height="24"
                        stroke="currentColor"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-width="2"
                        viewBox="0 0 24 24"
                        width="24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <rect
                          height="5"
                          rx="1"
                          width="20"
                          x="2"
                          y="3"
                        />
                        <path
                          d="M4 8v11a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2V8"
                        />
                        <path
                          d="M10 12h4"
                        />
                      </svg>
                    </button>
                  </div>
                </div>
              </div>
              <div
                class="text-sm text-gray-700 mb-2 line-clamp-2"
              >
                Test summary
              </div>
              <div
                class="flex items-center justify-between mt-2"
              >
                <div
                  class="flex flex-wrap gap-1"
                />
                <span
                  class="text-xs text-gray-400 ml-2 whitespace-nowrap"
                >
                  1/1/2023
                   ·
                   
                  1
                   min read
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="rounded-lg p-4 flex items-start cursor-pointer transition-all duration-200 bg-white hover:bg-neutral-50  border border-neutral-200"
      >
        <div
          class="flex-1 min-w-0"
        >
          <div
            class="flex items-start gap-3 mb-1"
          >
            <img
              alt="Test Newsletter 2"
              class="w-10 h-10 rounded object-cover bg-gray-100 flex-shrink-0 mt-1"
              src="https://example.com/image.jpg"
            />
            <div
              class="flex-1 min-w-0"
            >
              <div
                class="flex items-start justify-between gap-3"
              >
                <div
                  class="flex-1 min-w-0"
                >
                  <div
                    class="font-semibold text-base truncate"
                  >
                    Test Newsletter 2
                  </div>
                  <div
                    class="text-sm text-gray-500 truncate"
                  >
                    Test Source 2
                    <span
                      class="text-gray-400 ml-2"
                    >
                      • 
                      test2@example.com
                    </span>
                  </div>
                </div>
                <div
                  class="flex items-center gap-1 flex-shrink-0"
                >
                  <button
                    class="btn btn-ghost btn-xs p-1.5 rounded-lg hover:bg-gray-200 transition-colors "
                    title="Edit tags"
                    type="button"
                  >
                    <svg
                      class="lucide lucide-tag text-gray-500 hover:text-primary-600"
                      fill="none"
                      height="14"
                      stroke="currentColor"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="2"
                      viewBox="0 0 24 24"
                      width="14"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M12.586 2.586A2 2 0 0 0 11.172 2H4a2 2 0 0 0-2 2v7.172a2 2 0 0 0 .586 1.414l8.704 8.704a2.426 2.426 0 0 0 3.42 0l6.58-6.58a2.426 2.426 0 0 0 0-3.42z"
                      />
                      <circle
                        cx="7.5"
                        cy="7.5"
                        fill="currentColor"
                        r=".5"
                      />
                    </svg>
                  </button>
                  <div
                    class="flex items-center gap-0.5"
                  >
                    <button
                      class="btn btn-ghost btn-xs p-1.5 rounded-lg hover:bg-gray-200 transition-all "
                      title="Mark as unread"
                      type="button"
                    >
                      <svg
                        class="lucide lucide-eye-off h-3.5 w-3.5 text-gray-600"
                        fill="none"
                        height="24"
                        stroke="currentColor"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-width="2"
                        viewBox="0 0 24 24"
                        width="24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M9.88 9.88a3 3 0 1 0 4.24 4.24"
                        />
                        <path
                          d="M10.73 5.08A10.43 10.43 0 0 1 12 5c7 0 10 7 10 7a13.16 13.16 0 0 1-1.67 2.68"
                        />
                        <path
                          d="M6.61 6.61A13.526 13.526 0 0 0 2 12s3 7 10 7a9.74 9.74 0 0 0 5.39-1.61"
                        />
                        <line
                          x1="2"
                          x2="22"
                          y1="2"
                          y2="22"
                        />
                      </svg>
                    </button>
                    <button
                      class="btn btn-ghost btn-xs p-1.5 rounded-lg hover:bg-gray-200 transition-all "
                      title="Like"
                      type="button"
                    >
                      <svg
                        class="lucide lucide-heart h-3.5 w-3.5 text-gray-600 hover:text-red-500"
                        fill="none"
                        height="24"
                        stroke="currentColor"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-width="2"
                        viewBox="0 0 24 24"
                        width="24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M19 14c1.49-1.46 3-3.21 3-5.5A5.5 5.5 0 0 0 16.5 3c-1.76 0-3 .5-4.5 2-1.5-1.5-2.74-2-4.5-2A5.5 5.5 0 0 0 2 8.5c0 2.3 1.5 4.05 3 5.5l7 7Z"
                        />
                      </svg>
                    </button>
                    <button
                      class="btn btn-ghost btn-xs p-1.5 rounded-lg hover:bg-gray-200 transition-all "
                      title="Add to queue"
                      type="button"
                    >
                      <svg
                        class="lucide lucide-bookmark h-3.5 w-3.5 text-gray-600 hover:text-yellow-500"
                        fill="none"
                        height="24"
                        stroke="currentColor"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-width="2"
                        viewBox="0 0 24 24"
                        width="24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="m19 21-7-4-7 4V5a2 2 0 0 1 2-2h10a2 2 0 0 1 2 2v16z"
                        />
                      </svg>
                    </button>
                    <button
                      class="btn btn-ghost btn-xs p-1.5 rounded-lg hover:bg-gray-200 transition-all "
                      title="Archive"
                      type="button"
                    >
                      <svg
                        class="lucide lucide-archive h-3.5 w-3.5 text-amber-700"
                        fill="none"
                        height="24"
                        stroke="currentColor"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-width="2"
                        viewBox="0 0 24 24"
                        width="24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <rect
                          height="5"
                          rx="1"
                          width="20"
                          x="2"
                          y="3"
                        />
                        <path
                          d="M4 8v11a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2V8"
                        />
                        <path
                          d="M10 12h4"
                        />
                      </svg>
                    </button>
                  </div>
                </div>
              </div>
              <div
                class="text-sm text-gray-700 mb-2 line-clamp-2"
              >
                Test summary
              </div>
              <div
                class="flex items-center justify-between mt-2"
              >
                <div
                  class="flex flex-wrap gap-1"
                />
                <span
                  class="text-xs text-gray-400 ml-2 whitespace-nowrap"
                >
                  1/2/2023
                   ·
                   
                  1
                   min read
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div>
      <div
        class="flex flex-col items-center justify-center py-8 text-center mt-4"
      >
        <div
          class="text-gray-400 mb-2"
        >
          <svg
            class="w-8 h-8 mx-auto"
            fill="none"
            stroke="currentColor"
            viewBox="0 0 24 24"
          >
            <path
              d="M5 13l4 4L19 7"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </svg>
        </div>
        <p
          class="text-sm text-gray-500 font-medium"
        >
          All 2 newsletters loaded
        </p>
        <p
          class="text-xs text-gray-400 mt-1"
        >
          Showing 
          2
           of 
          2
           newsletters
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`InfiniteNewsletterList > shows loading more indicator when fetching next page 1`] = `
<div>
  <div
    class=""
  >
    <div
      class="space-y-0 divide-y divide-gray-100"
    >
      <div
        class="rounded-lg p-4 flex items-start cursor-pointer transition-all duration-200 bg-blue-50/60 border-l-3 border-blue-500 hover:bg-blue-100/50  border border-neutral-200"
      >
        <div
          class="flex-1 min-w-0"
        >
          <div
            class="flex items-start gap-3 mb-1"
          >
            <img
              alt="Test Newsletter 1"
              class="w-10 h-10 rounded object-cover bg-gray-100 flex-shrink-0 mt-1"
              src="https://example.com/image.jpg"
            />
            <div
              class="flex-1 min-w-0"
            >
              <div
                class="flex items-start justify-between gap-3"
              >
                <div
                  class="flex-1 min-w-0"
                >
                  <div
                    class="font-semibold text-base truncate"
                  >
                    Test Newsletter 1
                  </div>
                  <div
                    class="text-sm text-gray-500 truncate"
                  >
                    Test Source
                    <span
                      class="text-gray-400 ml-2"
                    >
                      • 
                      test@example.com
                    </span>
                  </div>
                </div>
                <div
                  class="flex items-center gap-1 flex-shrink-0"
                >
                  <button
                    class="btn btn-ghost btn-xs p-1.5 rounded-lg hover:bg-gray-200 transition-colors "
                    title="Edit tags"
                    type="button"
                  >
                    <svg
                      class="lucide lucide-tag text-gray-500 hover:text-primary-600"
                      fill="none"
                      height="14"
                      stroke="currentColor"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="2"
                      viewBox="0 0 24 24"
                      width="14"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M12.586 2.586A2 2 0 0 0 11.172 2H4a2 2 0 0 0-2 2v7.172a2 2 0 0 0 .586 1.414l8.704 8.704a2.426 2.426 0 0 0 3.42 0l6.58-6.58a2.426 2.426 0 0 0 0-3.42z"
                      />
                      <circle
                        cx="7.5"
                        cy="7.5"
                        fill="currentColor"
                        r=".5"
                      />
                    </svg>
                  </button>
                  <div
                    class="flex items-center gap-0.5"
                  >
                    <button
                      class="btn btn-ghost btn-xs p-1.5 rounded-lg hover:bg-gray-200 transition-all "
                      title="Mark as read"
                      type="button"
                    >
                      <svg
                        class="lucide lucide-eye h-3.5 w-3.5 text-blue-600"
                        fill="none"
                        height="24"
                        stroke="currentColor"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-width="2"
                        viewBox="0 0 24 24"
                        width="24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M2 12s3-7 10-7 10 7 10 7-3 7-10 7-10-7-10-7Z"
                        />
                        <circle
                          cx="12"
                          cy="12"
                          r="3"
                        />
                      </svg>
                    </button>
                    <button
                      class="btn btn-ghost btn-xs p-1.5 rounded-lg hover:bg-gray-200 transition-all "
                      title="Like"
                      type="button"
                    >
                      <svg
                        class="lucide lucide-heart h-3.5 w-3.5 text-gray-600 hover:text-red-500"
                        fill="none"
                        height="24"
                        stroke="currentColor"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-width="2"
                        viewBox="0 0 24 24"
                        width="24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M19 14c1.49-1.46 3-3.21 3-5.5A5.5 5.5 0 0 0 16.5 3c-1.76 0-3 .5-4.5 2-1.5-1.5-2.74-2-4.5-2A5.5 5.5 0 0 0 2 8.5c0 2.3 1.5 4.05 3 5.5l7 7Z"
                        />
                      </svg>
                    </button>
                    <button
                      class="btn btn-ghost btn-xs p-1.5 rounded-lg hover:bg-gray-200 transition-all "
                      title="Add to queue"
                      type="button"
                    >
                      <svg
                        class="lucide lucide-bookmark h-3.5 w-3.5 text-gray-600 hover:text-yellow-500"
                        fill="none"
                        height="24"
                        stroke="currentColor"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-width="2"
                        viewBox="0 0 24 24"
                        width="24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="m19 21-7-4-7 4V5a2 2 0 0 1 2-2h10a2 2 0 0 1 2 2v16z"
                        />
                      </svg>
                    </button>
                    <button
                      class="btn btn-ghost btn-xs p-1.5 rounded-lg hover:bg-gray-200 transition-all "
                      title="Archive"
                      type="button"
                    >
                      <svg
                        class="lucide lucide-archive h-3.5 w-3.5 text-amber-700"
                        fill="none"
                        height="24"
                        stroke="currentColor"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-width="2"
                        viewBox="0 0 24 24"
                        width="24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <rect
                          height="5"
                          rx="1"
                          width="20"
                          x="2"
                          y="3"
                        />
                        <path
                          d="M4 8v11a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2V8"
                        />
                        <path
                          d="M10 12h4"
                        />
                      </svg>
                    </button>
                  </div>
                </div>
              </div>
              <div
                class="text-sm text-gray-700 mb-2 line-clamp-2"
              >
                Test summary
              </div>
              <div
                class="flex items-center justify-between mt-2"
              >
                <div
                  class="flex flex-wrap gap-1"
                />
                <span
                  class="text-xs text-gray-400 ml-2 whitespace-nowrap"
                >
                  1/1/2023
                   ·
                   
                  1
                   min read
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="rounded-lg p-4 flex items-start cursor-pointer transition-all duration-200 bg-white hover:bg-neutral-50  border border-neutral-200"
      >
        <div
          class="flex-1 min-w-0"
        >
          <div
            class="flex items-start gap-3 mb-1"
          >
            <img
              alt="Test Newsletter 2"
              class="w-10 h-10 rounded object-cover bg-gray-100 flex-shrink-0 mt-1"
              src="https://example.com/image.jpg"
            />
            <div
              class="flex-1 min-w-0"
            >
              <div
                class="flex items-start justify-between gap-3"
              >
                <div
                  class="flex-1 min-w-0"
                >
                  <div
                    class="font-semibold text-base truncate"
                  >
                    Test Newsletter 2
                  </div>
                  <div
                    class="text-sm text-gray-500 truncate"
                  >
                    Test Source 2
                    <span
                      class="text-gray-400 ml-2"
                    >
                      • 
                      test2@example.com
                    </span>
                  </div>
                </div>
                <div
                  class="flex items-center gap-1 flex-shrink-0"
                >
                  <button
                    class="btn btn-ghost btn-xs p-1.5 rounded-lg hover:bg-gray-200 transition-colors "
                    title="Edit tags"
                    type="button"
                  >
                    <svg
                      class="lucide lucide-tag text-gray-500 hover:text-primary-600"
                      fill="none"
                      height="14"
                      stroke="currentColor"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="2"
                      viewBox="0 0 24 24"
                      width="14"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M12.586 2.586A2 2 0 0 0 11.172 2H4a2 2 0 0 0-2 2v7.172a2 2 0 0 0 .586 1.414l8.704 8.704a2.426 2.426 0 0 0 3.42 0l6.58-6.58a2.426 2.426 0 0 0 0-3.42z"
                      />
                      <circle
                        cx="7.5"
                        cy="7.5"
                        fill="currentColor"
                        r=".5"
                      />
                    </svg>
                  </button>
                  <div
                    class="flex items-center gap-0.5"
                  >
                    <button
                      class="btn btn-ghost btn-xs p-1.5 rounded-lg hover:bg-gray-200 transition-all "
                      title="Mark as unread"
                      type="button"
                    >
                      <svg
                        class="lucide lucide-eye-off h-3.5 w-3.5 text-gray-600"
                        fill="none"
                        height="24"
                        stroke="currentColor"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-width="2"
                        viewBox="0 0 24 24"
                        width="24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M9.88 9.88a3 3 0 1 0 4.24 4.24"
                        />
                        <path
                          d="M10.73 5.08A10.43 10.43 0 0 1 12 5c7 0 10 7 10 7a13.16 13.16 0 0 1-1.67 2.68"
                        />
                        <path
                          d="M6.61 6.61A13.526 13.526 0 0 0 2 12s3 7 10 7a9.74 9.74 0 0 0 5.39-1.61"
                        />
                        <line
                          x1="2"
                          x2="22"
                          y1="2"
                          y2="22"
                        />
                      </svg>
                    </button>
                    <button
                      class="btn btn-ghost btn-xs p-1.5 rounded-lg hover:bg-gray-200 transition-all "
                      title="Like"
                      type="button"
                    >
                      <svg
                        class="lucide lucide-heart h-3.5 w-3.5 text-gray-600 hover:text-red-500"
                        fill="none"
                        height="24"
                        stroke="currentColor"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-width="2"
                        viewBox="0 0 24 24"
                        width="24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M19 14c1.49-1.46 3-3.21 3-5.5A5.5 5.5 0 0 0 16.5 3c-1.76 0-3 .5-4.5 2-1.5-1.5-2.74-2-4.5-2A5.5 5.5 0 0 0 2 8.5c0 2.3 1.5 4.05 3 5.5l7 7Z"
                        />
                      </svg>
                    </button>
                    <button
                      class="btn btn-ghost btn-xs p-1.5 rounded-lg hover:bg-gray-200 transition-all "
                      title="Add to queue"
                      type="button"
                    >
                      <svg
                        class="lucide lucide-bookmark h-3.5 w-3.5 text-gray-600 hover:text-yellow-500"
                        fill="none"
                        height="24"
                        stroke="currentColor"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-width="2"
                        viewBox="0 0 24 24"
                        width="24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="m19 21-7-4-7 4V5a2 2 0 0 1 2-2h10a2 2 0 0 1 2 2v16z"
                        />
                      </svg>
                    </button>
                    <button
                      class="btn btn-ghost btn-xs p-1.5 rounded-lg hover:bg-gray-200 transition-all "
                      title="Archive"
                      type="button"
                    >
                      <svg
                        class="lucide lucide-archive h-3.5 w-3.5 text-amber-700"
                        fill="none"
                        height="24"
                        stroke="currentColor"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-width="2"
                        viewBox="0 0 24 24"
                        width="24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <rect
                          height="5"
                          rx="1"
                          width="20"
                          x="2"
                          y="3"
                        />
                        <path
                          d="M4 8v11a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2V8"
                        />
                        <path
                          d="M10 12h4"
                        />
                      </svg>
                    </button>
                  </div>
                </div>
              </div>
              <div
                class="text-sm text-gray-700 mb-2 line-clamp-2"
              >
                Test summary
              </div>
              <div
                class="flex items-center justify-between mt-2"
              >
                <div
                  class="flex flex-wrap gap-1"
                />
                <span
                  class="text-xs text-gray-400 ml-2 whitespace-nowrap"
                >
                  1/2/2023
                   ·
                   
                  1
                   min read
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div>
      <div
        class="flex flex-col items-center justify-center py-8 mt-4"
      >
        <svg
          class="lucide lucide-loader2 w-6 h-6 animate-spin text-blue-500 mb-3"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M21 12a9 9 0 1 1-6.219-8.56"
          />
        </svg>
        <p
          class="text-sm text-gray-500 font-medium"
        >
          Loading more newsletters...
        </p>
        <p
          class="text-xs text-gray-400 mt-1"
        >
          Loaded 
          2
           of 
          2
        </p>
      </div>
    </div>
  </div>
</div>
`;
